src/main/java/org/opentripplanner/middleware/models/AbstractUser.java

```diff
@@ -28,8 +28,11 @@
     /**
      * Set of permissions the user has.
      * TODO: fill out permissions more completely.
+     * NOTE: permissions is marked as transient to prevent serialization by GSON that results in an exception
+     * trying to serialize Java.class inside {@link Permission} and a 500 error when querying users from UI.
+     * (See, for instance, https://howtoprogram.xyz/2016/10/16/ignore-or-exclude-field-in-gson/ for more info.)
      */
-    public Set<Permission> permissions = new HashSet<>();
+    public transient Set<Permission> permissions = new HashSet<>();
```

Comment:
I don't think this is a GSON error, but rather a MongoDB error. Please add FIXME.

Reply:
How about changing the visibility to `protected` because we only use this field in `AbstractUser` and `AdminUser`? That also prevents serialization and we don't need a special syntax .